README.md

```diff
@@ -1,57 +1,202 @@
+    ```
+
+  The emulator exposes several ports: 
+  - `8080`: gRPC endpoint that allows the app to connect to the scheduler
+  - `8082`: endpoint for monitoring dashboard  
+
+   ![Docker desktop view of emulator container](./media/emulator-container.png)
+
+   Use the port number mapped to `8080` in your durable task scheduler connection string. For example, in this case, the connection string would be `Endpoint=http://localhost:55000>;Authentication=None`
+
+### Run app using Visual Studio Code
+
+1. Open **app** folder in a new terminal 
+2. Open VS Code by entering `code .` in the terminal
+3. In the root folder, create a file named `local.settings.json` with the following, filling in connection information from the previous step: 
+    ```json
+    {
+      "IsEncrypted": false,
+      "Values": {
+          "AzureWebJobsStorage": "UseDevelopmentStorage=true",
+          "MODEL_ENDPOINT": "https://<resource name>.services.ai.azure.com/models",
+          "MODEL_API_KEY": "<api key>", 
+          "DURABLE_TASK_SCHEDULER_CONNECTION_STRING": "Endpoint=http://localhost:<port number>;Authentication=None",
+          "TASKHUB_NAME": "default", 
+          "FUNCTIONS_WORKER_RUNTIME": "python"
+      }
+    }
+    ```
+4. Start Azurite by opening the command template and searching for `Azurite: Start`
+
+5. Run project with debugging (or press F5)
+
+6. You can test easily by going to the `test.http` file and click "Send Request". This file has POST requests asking different questions. For example: 
+  
+    *"What is the value proposition of Durable Functions and what is it used for?"*
+
+    The request will return an HTTP response with some URLs that allow you to manage the orchestration, but this sample won't be using those.
+
+7. Check the `mylog.log` file. [This file](./app/mylog.log) logs the prompt and response from each language model, as well as the final evaluation result. 
+
+### Inspect the solution 
+
+Take a look at the `orchestrator_function` to see how Durable Functions allows you to write code that runs in parallel. This function simply adds the activity functions that make calls to language models to a list and then call `context.task_all(tasks)`, which would signal the activity functions to run in parallel. Note that you don't have to worry about when each activity functions finishes or if any fail in the middle - Durable Functions handles the "fan in" and the automatic retries. Simply take the result and continue with your business logic.
+
+```python
+@app.orchestration_trigger(context_name="context")
+def orchestrator_function(context):
+  # Previous logic
+  
+  tasks = [
+    context.call_activity_with_retry("get_gpt35_result", retry_options, [user_prompt, system_prompt]),
+    context.call_activity_with_retry("get_gpt4omini_result", retry_options, [user_prompt, system_prompt]),
+    context.call_activity_with_retry("get_phi4_result", retry_options, [user_prompt, system_prompt])
+  ]
+  
+  # Run all tasks in parallel
```

Comment:
This comment should be moved up, above the point where we initialize the `tasks` array. We should then add a common here that says `# Wait for all the parallel tasks to complete before continuing`.